services/nats/pkg/server/nats/nats.go

```diff
@@ -44,6 +45,8 @@ func (n *NATSServer) ListenAndServe() (err error) {
 	return nil
 }
 
+// Shutdown stops the NATSServer gracefully
+// The NATS server itself runs the signal handling. Set `natsOpts.NoSigs = true` if you want to handle signals ourselves
```

Comment:
I think we should use the `natsOpts.NoSigs = true` option. The nats signal handler calls os.Exit() which might cause an unclean shutdown for the rest of ocis when running as a single binary.

I guess you'll have to add a n.server.WaitForShutdown() call below as well then.

Reply:
Good point!